### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by abhishektanwar**
Added login and signup functionalities.

Files to check:
- src/App.tsx
- src/components/Authentication/Login.tsx
- src/components/Authentication/SignUp.tsx
- src/components/InputField/index.tsx 
- src/components/ModalWrapper/index.tsx
- src/contexts/AuthDialogContext/index.tsx
- src/contexts/ModalContext/index.tsx
- src/utils/useAxios/index.tsx

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+import logo from "./logo.svg";
+import "./App.css";
+import Header from "./components/Header";
+import { ModalProvider } from "./contexts/ModalContext";
+import { AuthProvider } from "./contexts/AuthDialogContext";
+import { Loader } from "./components/Loader";
+import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
+
 
 function App() {
```
Comment (iRohitGaur) [inline src/App.tsx:11]: suggestion: import `FC` from "react" and make them the standard functional components in ts

File: src/components/Authentication/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { useAuth } from "../../contexts/AuthDialogContext";
+import { useModal } from "../../contexts/ModalContext";
+import Button from "../Header/Button";
+import InputField from "../InputField";
+import {Loader} from '../../components/Loader';
+import "./authentication.css";
+const Login = () => {
+  // const { setAuthType, loginHandler, error } = useAuth();
```
Comment (iRohitGaur) [inline src/components/Authentication/Login.tsx:9]: suggestion: remove commented code

**Review by srishtigautam21 — APPROVED** [review]
neatly structured code
Looks good to merge

File: src/components/Authentication/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { useAuth } from "../../contexts/AuthDialogContext";
+import { useModal } from "../../contexts/ModalContext";
```
Comment (iRohitGaur) [inline src/components/Authentication/Login.tsx:3]: suggestion: use an index file in `contexts` and export all the files of that folder from there to get cleaner imports

File: src/components/Authentication/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  const handleLoginCredentialInput = (e:any) => {
+    setLoginCredentials((creds) => ({
+      ...creds,
+      [e.target.name]: e.target.value,
+    }));
+  };
+
+  const loginUser = async () => {
+    // e.preventDefault();
```
Comment (iRohitGaur) [inline src/components/Authentication/Login.tsx:37]: suggestion: remove commented code

File: src/components/Authentication/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          <Button
+            buttonText="Create new account"
+            buttonStyle="secondary-button margin-top-0"
+            icon="fas fa-chevron-right"
+            onClick={() => setAuthType("signup")}
+          />
+          <Button
+            buttonText="Login with test credenials"
+            buttonStyle="secondary-button body-typo-md margin-top-0"
+            // icon="fas fa-chevron-right"
```
Comment (iRohitGaur) [inline src/components/Authentication/Login.tsx:90]: suggestion: remove commented code

File: src/components/Authentication/SignUp.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+            required={true}
+            validation={signUpCredentials.confirmPassword.length === 0 ? true : isValidPassword}
+            showTogglePasswordButton = {true}
+          />
+          <div className="card-action-btn-container flex-column">
+            <Button buttonText="Sign Up" buttonStyle={!(isValidPassword && isValidEmail) ? 'btn-disabled' : '' } onClick={() => signUpUser()} />
+            <Button
+              buttonText="Already have an account ?"
+              buttonStyle="secondary-button"
+              // icon="fas fa-chevron-right"
```
Comment (iRohitGaur) [inline src/components/Authentication/SignUp.tsx:96]: suggestion: remove commented code

File: src/components/Header/BadgeIconButton.jsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+              {badgeNumber}
+            </span>
+          ) : null}
+        </span>
+      </button>
+    </div>
+  );
+};
+
+export default BadgeIconButton;
```
Comment (iRohitGaur) [inline src/components/Header/BadgeIconButton.jsx:21]: suggestion: keep all files in `.tsx` for consistency

File: src/components/Header/Logo.jsx [hunk 8]
```diff
@@ -0,0 +1,150 @@
+import React from "react";
+const Logo = () => {
+  return (
+    <div className="nav-logo nav-section-items">
+      {/* <Link to="/" className="link-no-style" > */}
+      {/* <a className="link-no-style" href="/"> */}
```
Comment (iRohitGaur) [inline src/components/Header/Logo.jsx:6]: suggestion: remove commented code

File: src/components/Header/index.jsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
+import React from "react";
+// import { Link } from "react-router-dom";
+import { useAuth } from "../../contexts/AuthDialogContext";
+import { useModal } from "../../contexts/ModalContext";
+// import { useWishlist } from "../../Contexts/WishlistContext";
```
Comment (iRohitGaur) [inline src/components/Header/index.jsx]: suggestion: remove commented code

File: src/components/Header/index.jsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+  };
+  return (
+    <>
+      {/* <Link to="/">
+          <Logo />
+        <span className="show-mobile-logo">
+          <MobileLogo />
+        </span>
+      </Link> */}
+      {/* <SearchBar /> */}
```
Comment (iRohitGaur) [inline src/components/Header/index.jsx:26]: suggestion: remove commented code

File: src/utils/useAxios/index.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { useState } from "react"
+import axios from 'axios';
+import { useToast } from "../../hooks/useToast";
+
+axios.defaults.baseURL = "";
+
+export const useAxios = () => {
+  
+  const operation = async (params:any) => {
```
Comment (iRohitGaur) [inline src/utils/useAxios/index.tsx]: suggestion: `params` should be of type `AxiosRequestConfig`

File: src/components/Header/Header.css [hunk 12]
```diff
@@ -0,0 +1,78 @@
+.nav-bar{
+  padding:unset;
+}
+
+.search-bar-container{
+  width: 50%;
+  border: 1px solid #D1D1D1;
+  padding: 10px;
+  border-radius: 5px;
+  background-color: #F9F9F9;
```
Comment (iRohitGaur) [inline src/components/Header/Header.css:10]: suggestion: use variables

File: src/components/Header/Header.css [hunk 13]
```diff
@@ -0,0 +1,78 @@
+.nav-bar{
+  padding:unset;
+}
+
+.search-bar-container{
+  width: 50%;
+  border: 1px solid #D1D1D1;
```
Comment (iRohitGaur) [inline src/components/Header/Header.css:7]: suggestion: use variables

File: src/components/ModalWrapper/modal-wrapper.css [hunk 14]
```diff
@@ -0,0 +1,28 @@
+.modal-background{
+  width: 100%;
+  height: 100%;
+  background:rgba(0,0,0,0.1);
```
Comment (iRohitGaur) [inline src/components/ModalWrapper/modal-wrapper.css:4]: suggestion: use variables

File: src/components/ModalWrapper/modal-wrapper.css [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 20;
+}
+
+.modal-wrapper{
+  width: 80rem;
+  height: 50rem;
+  box-shadow: 0 5px 16px rgba(0,0,0,0.2);
```
Comment (iRohitGaur) [inline src/components/ModalWrapper/modal-wrapper.css:15]: suggestion: use variables

**Review by iRohitGaur — APPROVED** [review]
UI looks nice.

Gave some suggestions, please look into it and do the modification wherever needed.

Good to merge with changes 👍🏻

File: src/App.tsx (hunk 1 above)
Comment (abhishektanwar) [reply src/App.tsx:11]: fixed. thanks for mentioning it

File: src/components/Authentication/Login.tsx (hunk 2 above)
Comment (abhishektanwar) [reply src/components/Authentication/Login.tsx:9]: removed

File: src/components/Authentication/Login.tsx (hunk 4 above)
Comment (abhishektanwar) [reply src/components/Authentication/Login.tsx:37]: fixed

File: src/components/Authentication/Login.tsx (hunk 5 above)
Comment (abhishektanwar) [reply src/components/Authentication/Login.tsx:90]: fixed.

File: src/components/Authentication/SignUp.tsx (hunk 6 above)
Comment (abhishektanwar) [reply src/components/Authentication/SignUp.tsx:96]: fixed.

File: src/components/Header/BadgeIconButton.jsx (hunk 7 above)
Comment (abhishektanwar) [reply src/components/Header/BadgeIconButton.jsx:21]: fixed

File: src/components/Header/Logo.jsx (hunk 8 above)
Comment (abhishektanwar) [reply src/components/Header/Logo.jsx:6]: This will be fixed in next PR. This file would be removed.

File: src/components/Header/index.jsx (hunk 9 above)
Comment (abhishektanwar) [reply src/components/Header/index.jsx]: fixed

File: src/utils/useAxios/index.tsx (hunk 11 above)
Comment (abhishektanwar) [reply src/utils/useAxios/index.tsx]: fixed. Thank for mentioning it.

**Pull request merged**